ludwig/encoders/text_encoders.py

```diff
@@ -59,6 +60,48 @@ def _cls_pooled_error_message(encoder: str):
 class HFTextEncoder(Encoder):
     DEFAULT_MODEL_NAME: str
 
+    def _init_config(self, transformer, schema_keys: List[str], encoder_config: SequenceEncoderConfig):
+        """Creates a config object for the encoder using the transformer model and the passed-in encoder config.
+
+        The transformer's config is only known after it is instantiated, so we must update the
+        encoder config with the values from the transformer config.
+
+        Args:
+            transformer: The transformer model.
+            schema_keys: The keys in the encoder config schema. We only want to update the encoder config
+                with the values from the transformer config that are in the schema.
+            encoder_config: The existing encoder config containing defaults and user-specified values.
+                If the values in this config differ from the transformer's config, the transformer's config
+                values will override this config's values.
+        Returns:
+            A new encoder config object with the updated values from the transformer config.
+        """
+        transformer_config = transformer.config.to_dict()
+        final_hf_config_params = {k: v for k, v in transformer_config.items() if k in schema_keys}
+        encoder_config_dict = encoder_config.to_dict()
+        encoder_config_dict.update(final_hf_config_params)
+        return self.get_schema_cls().from_dict(encoder_config_dict)
+
+    def _maybe_resize_token_embeddings(self, transformer, vocab_size: int):
+        """Resizes the token embeddings if the vocab size is different from the transformer's vocab size.
+
+        This should only happen if we are instantiating a model from scratch (i.e. not loading from a pretrained model
+        or checkpoint). Pretrained models update the vocab size stored in the config. This means if we are loading a
+        pretrained model from a checkpoint, the config vocab size should match the model's vocab size.
+
+        It is important that pretrained models update the vocab size stored in the config because sometimes the
+        pretrained models will have an embeddings table that is a different size than the vocab size. Examples:
+
+        CamemBERT:  https://github.com/huggingface/tokenizers/issues/900#issue-1122256698
+        T5:         https://github.com/huggingface/transformers/issues/4875#issue-635471552
+
+        Args:
+            transformer: The transformer model.
+            vocab_size: The vocab size of the dataset.
+        """
+        if vocab_size != transformer.config.vocab_size:
```

Comment:
shouldn't we be doing the opposite? use wathever vocab the transformer is telling us to use?

Reply:
Yup, this method is only called if we are instantiating the model from scratch (i.e. `use_pretrained==False`). I've included this as part of the docstring of this method [here](https://github.com/ludwig-ai/ludwig/pull/3010/files#diff-f34cc840e21b57eebb191a5198da3fce8ce7dc5278fa4ea404d8d960a23f303dR88-R90), and you can see an example of this in one of the text encoders [here](https://github.com/ludwig-ai/ludwig/pull/3010/files#diff-f34cc840e21b57eebb191a5198da3fce8ce7dc5278fa4ea404d8d960a23f303dR415-R421). 

If `use_pretrained==True`, then this function is not called and instead we directly use the token embeddings table determined by the transformer.